### PR TITLE
Fixed failed calls on HPC with backup FS server

### PR DESCRIPTION
**Pull request by sergey-safarov**

### CODE REVIEW

File: kamailio/dispatcher-role.cfg [hunk 1]
```diff
@@ -86,6 +86,7 @@ route[DISPATCHER_FIND_ROUTES]
     #!import_file "dispatcher-network-find.cfg"
 
     $var(ds_group) = $var(ds_primary_group);
+    $avp(ds_group) = $var(ds_group);
```
Comment (sergey-safarov) [inline kamailio/dispatcher-role.cfg]: I can replace `$var(ds_group)` to `$avp(ds_group)` on whole file

File: kamailio/dispatcher-role.cfg [hunk 2]
```diff
@@ -171,27 +172,35 @@ route[DISPATCHER_NEXT_ROUTE]
 
     if ($sht(failover=>$ci::counts) == $null) {
         $var(i) = 0;
+        $(avp(ds_dst)[$var(i)]) = $null; # Removing already failed dispatcher route
+
+        # Also try backup servers
+        if ($avp(ds_group) == 1 && ds_select_dst("2", "0")) {
```
Comment (sergey-safarov) [inline kamailio/dispatcher-role.cfg]: When #91 and #92 is merged, then i change `1` and `2` to new macros.

File: kamailio/dispatcher-role.cfg [hunk 3]
```diff
@@ -171,27 +172,35 @@ route[DISPATCHER_NEXT_ROUTE]
...
+            xlog("L_CRIT", "$ci|log|Trying deliver call via backup dispatcher servers\n");
+        }
+
         while($(avp(ds_dst)[$var(i)]) != $null) {
             $sht(failover=>$ci[$var(i)]) = $(avp(ds_dst)[$var(i)]);
             $var(i) = $var(i) + 1;
-            if ($var(i) >= 3) {
+            if ($var(i) >= 2) {
```
Comment (sergey-safarov) [inline kamailio/dispatcher-role.cfg]: Here `2` also may be changed to macro

File: kamailio/dispatcher-role.cfg [hunk 4]
```diff
@@ -171,27 +172,35 @@ route[DISPATCHER_NEXT_ROUTE]
...
         }
+
         $sht(failover=>$ci::counts) = $var(i);
         $var(failover_count) = $var(i);
     }
 
     # try to find a new media server to send the call to
-    if ($var(failover_count) > 1) {
+    if ($var(failover_count) > 0) {
         $var(failover_count) = $var(failover_count) - 1;
```
Comment (sergey-safarov) [inline kamailio/dispatcher-role.cfg]: Later need to change failover media servers order from `0` to 'MAX_FAILOVER_ATTEMPTS`
In this case backup servers tried first. Now primary servers tied first if `0 < backup_server < MAX_FAILOVER_ATTEMPTS`

**Comment by lazedo** [comment]
@sergey-safarov dispatcher largely refactored. i believe you can achieve what your looking by providing custom implementations with the existing `import_file`

**Pull request closed without merging**